src/Http/Uri.php

```diff
@@ -0,0 +1,84 @@
+<?php
+declare(strict_types=1);
+
+namespace Cake\Http;
+
+use Laminas\Diactoros\Uri as LaminasUri;
+
+/**
+ * The base and webroot properties have piggybacked on the Uri for
+ * a long time. To preserve backwards compatibility and avoid dynamic
+ * property errors in PHP 8.2 we use this subclass.
+ */
+class Uri extends LaminasUri
+{
+    /**
+     * @var string
+     */
+    private $base = '';
+
+    /**
+     * @var string
+     */
+    private $webroot = '';
+
+    /**
+     * Create a subclassed Uri that includes additional CakePHP properties
+     *
+     * @param \Laminas\Diactoros\Uri $uri Uri instance to copy
+     * @param string $base The base path.
+     * @param string $webroot The webroot path.
+     * @return self
+     */
+    public static function fromLaminas(LaminasUri $uri, string $base, string $webroot): self
+    {
+        $copy = (new self())
+            ->withScheme($uri->getScheme())
+            ->withHost($uri->getHost())
+            ->withUserInfo($uri->getUserInfo())
+            ->withPort($uri->getPort())
+            ->withPath($uri->getPath())
+            ->withQuery($uri->getQuery())
+            ->withFragment($uri->getFragment());
+
+        $copy->base = $base;
+        $copy->webroot = $webroot;
+
+        return $copy;
+    }
+
+    /**
+     * Backwards compatible property read
+     *
+     * @param string $prop The property to read.
+     * @return string|null
+     */
+    public function __get($prop): ?string
+    {
+        if ($prop !== 'base' && $prop !== 'webroot') {
+            return null;
```

Comment:
Is it convention to throw an exception here?

Reply:
trigger_error() would allow us to show a better error message.

Reply:
This is used the php docs example:
```php
        trigger_error(
            'Undefined property via __get(): ' . $name .
            ' in ' . $trace[0]['file'] .
            ' on line ' . $trace[0]['line'],
            E_USER_NOTICE);
```

Reply:
Yeah I'm not digging how this interface is working. I'm tempted to introduce a container for the Uri & our custom attributes and extract the Uri in the request. My hunch is that very few or no userland code is mucking with the guts of building a request.